classic_cp/N_Queens/N_Queens.jl

```diff
@@ -0,0 +1,168 @@
+- 'valueSelection': SeaPearl value selection. By default: =SeaPearl.BasicHeuristic()
+"""
+function model_queens(board_size::Int; benchmark=false, variableSelection=SeaPearl.MinDomainVariableSelection{false}(), valueSelection=SeaPearl.BasicHeuristic())
+    trailer = SeaPearl.Trailer()
+    model = SeaPearl.CPModel(trailer)
+
+    rows = Vector{SeaPearl.AbstractIntVar}(undef, board_size)
+    for i = 1:board_size
+        rows[i] = SeaPearl.IntVar(1, board_size, "row_"*string(i), trailer)
+        SeaPearl.addVariable!(model, rows[i]; branchable=true)
+    end
+
+    rows_plus = Vector{SeaPearl.AbstractIntVar}(undef, board_size)
+    for i = 1:board_size
+        rows_plus[i] = SeaPearl.IntVarViewOffset(rows[i], i, rows[i].id*"+"*string(i))
+    end
+
+    rows_minus = Vector{SeaPearl.AbstractIntVar}(undef, board_size)
+    for i = 1:board_size
+        rows_minus[i] = SeaPearl.IntVarViewOffset(rows[i], -i, rows[i].id*"-"*string(i))
+    end
+
+    push!(model.constraints, SeaPearl.AllDifferent(rows, trailer))
+    push!(model.constraints, SeaPearl.AllDifferent(rows_plus, trailer))
+    push!(model.constraints, SeaPearl.AllDifferent(rows_minus, trailer))
+
+    return model
+end
+
+"""
+    solve_queens(board_size::Int; benchmark=false, variableSelection=SeaPearl.MinDomainVariableSelection{false}(), valueSelection=SeaPearl.BasicHeuristic())
+
+Solve the SeaPearl model for to the N-Queens problem, using SeaPearl.MinDomainVariableSelection heuristique
+and  SeaPearl.AllDifferent, and the function model_queens
+
+# Arguments
+- `board_size::Int`: dimension of the board
+- 'variableSelection': SeaPearl variable selection. By default: SeaPearl.MinDomainVariableSelection{false}()
+- 'valueSelection': SeaPearl value selection. By default: =SeaPearl.BasicHeuristic()
+"""
+function solve_queens(board_size::Int; benchmark=false, variableSelection=SeaPearl.MinDomainVariableSelection{false}(), valueSelection=SeaPearl.BasicHeuristic())
+    model = model_queens(board_size::Int; benchmark=false, variableSelection=SeaPearl.MinDomainVariableSelection{false}(), valueSelection=SeaPearl.BasicHeuristic())
+    status = @time SeaPearl.solve!(model; variableHeuristic=variableSelection, valueSelection=valueSelection)
+    return model
+end
+
+"""
+    outputFromSeaPearl(model::SeaPearl.CPModel; optimality=false)
+
+Shows the results of the N-Queens problem as type OutputDataQueens.
+
+# Arguments
+- `model::SeaPearl.CPModel`: needs the model to be already solved (by solve_queens)
+"""
+function outputFromSeaPearl(model::SeaPearl.CPModel; optimality=false)
+    solutions = model.solutions
+    nb_sols = length(solutions)
+    board_size = length(model.variables)
+    indices = Matrix{Int}(undef,nb_sols, board_size)
+    for key in keys(solutions)
+        sol = solutions[key]
+        for i in 1:board_size
+            indices[key, i]= sol["row_"*string(i)]
+        end
+    end
```

Comment:
Matrices are stored by columns in Julia. It might be more coherent to store solutions by columns and not by rows, but this is a minor observation.